recipe/scripts/activate-dpcpp.sh

```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
+
+# This function takes no arguments
+# It tries to determine the name of this file in a programatic way.
+function _get_sourced_filename() {
+    if [ -n "${BASH_SOURCE[0]}" ]; then
+        basename "${BASH_SOURCE[0]}"
+    elif [ -n "${(%):-%x}" ]; then
+        # in zsh use prompt-style expansion to introspect the same information
+        # see http://stackoverflow.com/questions/9901210/bash-source0-equivalent-in-zsh
+        basename "${(%):-%x}"
+    else
+        echo "UNKNOWN FILE"
+    fi
+}
+
+if [ "${CONDA_BUILD:-0}" = "1" ]; then
+  if [ -f /tmp/old-env-$$.txt ]; then
+    rm -f /tmp/old-env-$$.txt || true
+  fi
+  env > /tmp/old-env-$$.txt
+fi
+
+if [ "${CC:-}" = "" ] && [ "${CONDA_BUILD:-}" = "1" ]; then
+  echo "Need to use C/C++ compiler activation package"
+  exit 1
+fi
+
+export CC=icx
+export CXX=icpx
+export ICPXCFG="$BUILD_PREFIX/bin/$HOST-icpx.cfg"
+export ICXCFG="$BUILD_PREFIX/bin/$HOST-icx.cfg"
+
+if [ $? -ne 0 ]; then
+  echo "ERROR: $(_get_sourced_filename) failed, see above for details"
+else
```

Comment:
How does the {{_get_sourced_filename}} get called? Forgive my scripting ignorance